src/clib/xtg/grd3d_point_val_crange.c

```diff
@@ -67,6 +67,10 @@ _find_ib(double x,
 
                 double corners[24];
                 ib = x_ijk2ib(i, j, k, nx, ny, nz, 0);
+                if (ib < 0) {
+                    throw_exception("Outside boundary");
+                    return EXIT_FAILURE;
+                }
```

Comment:
`/* cell not found -9; otherwise >= 0 but active may be 0 if inactive cell*/` ? It's not exactly the same issue though, as `-9` represents that `x, y, z` point was not found within any cells, while this exception throwing is because we are looking outside of the grid entirely.

Reply:
Thanks for a thorough review! So my previous return value was not a good option as a failure would be reported as a success. I changed it to return -8 if we are looking outside, acceptable?

Reply:
Yeah, I mean, I haven't seen an overview of the representation of the negative values, but at least `-8` would make more sense, considering how the function otherwise is used